Change `Screen.reading` pixel order to start from bottom left

jank324 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

The order of the pixels in `Screen.reading` was changed to start from the bottom left instead of the top left. This is now consistent with the `Screen.pixel_bin_centers` and `Screen.pixel_bin_edges` properties.

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
<!--- You can use the syntax `closes #100` if this solves the issue #100 -->

- [x] I have raised an issue to propose this change (required for new features and bug fixes)

Currently the order of the pixels in `reading` and `pixel_bin_centers` disagrees, which is confusing. Closes #395.

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

<!-- - [ ] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**) -->

- [x] I have updated the changelog accordingly (**required**).
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (_required for a bug fix or a new feature_).
- [ ] I have updated the documentation accordingly.
  <!-- - [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary) -->
  <!-- - [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary) -->
- [x] I have reformatted the code and checked that formatting passes (**required**).
- [x] I have have fixed all issues found by `flake8` (**required**).
- [x] I have ensured that all `pytest` tests pass (**required**).
- [ ] I have run `pytest` on a machine with a CUDA GPU and made sure all tests pass (**required**).
- [x] I have checked that the documentation builds (**required**).

Note: We are using a maximum length of 88 characters per line.

<!--- This Template is an edited version of the one from https://github.com/DLR-RM/stable-baselines3/ -->

jank324 commented:
For reference what this used to look like, please check the issue #395 associated with this PR.

The same code now results in the following:
```python
import matplotlib.pyplot as plt
import torch

import cheetah
```
... for a `ParameterBeam`:
```python
screen = cheetah.Screen(
    resolution=(500, 500),
    pixel_size=torch.ones(2) * 50e-6,
    is_active=True,
    method="histogram",
    kde_bandwidth=torch.tensor(50e-6),
)

beam = cheetah.ParticleBeam.from_twiss(
    beta_x=torch.tensor(1.0),
    beta_y=torch.tensor(10.0),
    alpha_x=torch.tensor(0.0),
    alpha_y=torch.tensor(0.0),
    emittance_x=torch.tensor(1e-9),
    emittance_y=torch.tensor(1e-7),
    total_charge=torch.tensor(1e-9),
)
beam.x += beam.y * 5.0
beam.x += 5e-3
beam.y += 2e-3
beam = beam.as_parameter_beam()

screen.track(beam)

beam.as_particle_beam(num_particles=10_000).plot_2d_distribution("x", "y")
plt.show()

plt.pcolor(*screen.pixel_bin_centers, screen.reading)
plt.xlabel("x (m)")
plt.ylabel("y (m)")
plt.show()
```
![Screenshot 2025-04-23 at 10 59 00](https://github.com/user-attachments/assets/4a234e8e-b87a-4e51-8eab-b4208d9de06d)
![Screenshot 2025-04-23 at 10 59 18](https://github.com/user-attachments/assets/a1cafe08-12db-4bb8-ab9d-f8d5c08329d3)
... for a `ParticleBeam` with `method="histogram"`:
```python
screen = cheetah.Screen(
    resolution=(500, 500),
    pixel_size=torch.ones(2) * 50e-6,
    is_active=True,
    method="histogram",
    kde_bandwidth=torch.tensor(50e-6),
)

beam = cheetah.ParticleBeam.from_twiss(
    beta_x=torch.tensor(1.0),
    beta_y=torch.tensor(10.0),
    alpha_x=torch.tensor(0.0),
    alpha_y=torch.tensor(0.0),
    emittance_x=torch.tensor(1e-9),
    emittance_y=torch.tensor(1e-7),
    num_particles=10_000,
    total_charge=torch.tensor(1e-9),
)
beam.x += beam.y * 5.0
beam.x += 5e-3
beam.y += 2e-3

screen.track(beam)

beam.plot_2d_distribution("x", "y")
plt.show()

plt.pcolor(*screen.pixel_bin_centers, screen.reading)
plt.xlabel("x (m)")
plt.ylabel("y (m)")
plt.show()
```
![Screenshot 2025-04-23 at 11 00 29](https://github.com/user-attachments/assets/c55a317b-5b23-4356-9bcf-559523d6d641)
![Screenshot 2025-04-23 at 11 00 45](https://github.com/user-attachments/assets/1ebb2f97-5a7d-4c54-9cc4-019db56aacdd)
... for a `ParticleBeam` with `method= "kde"`:
```python
screen = cheetah.Screen(
    resolution=(500, 500),
    pixel_size=torch.ones(2) * 50e-6,
    is_active=True,
    method="kde",
    kde_bandwidth=torch.tensor(50e-6),
)

beam = cheetah.ParticleBeam.from_twiss(
    beta_x=torch.tensor(1.0),
    beta_y=torch.tensor(10.0),
    alpha_x=torch.tensor(0.0),
    alpha_y=torch.tensor(0.0),
    emittance_x=torch.tensor(1e-9),
    emittance_y=torch.tensor(1e-7),
    num_particles=10_000,
    total_charge=torch.tensor(1e-9),
)
beam.x += beam.y * 5.0
beam.x += 5e-3
beam.y += 2e-3

screen.track(beam)

beam.plot_2d_distribution("x", "y")
plt.show()

plt.pcolor(*screen.pixel_bin_centers, screen.reading)
plt.xlabel("x (m)")
plt.ylabel("y (m)")
plt.show()
```
![Screenshot 2025-04-23 at 11 02 16](https://github.com/user-attachments/assets/2e3e3cce-514b-41c2-8128-46ea3d76843a)
![Screenshot 2025-04-23 at 11 01 59](https://github.com/user-attachments/assets/63fd2324-bf80-4a7e-97f1-7af4523c5e97)

jank324 commented:
@roussel-ryan can you check if this looks good now?

roussel-ryan commented:
why does the first call to `plot_2d_distribution` show a beam without any correlations between x and y?

jank324 commented:
> why does the first call to `plot_2d_distribution` show a beam without any correlations between x and y?

I was going to say because `ParameterBeam` doesn't take into account any off-diagonal entries in the covariance matrix, but I checked now and it actually does. So I'm a little confused about that too.

Hespe commented:
> > why does the first call to `plot_2d_distribution` show a beam without any correlations between x and y?
> 
> I was going to say because `ParameterBeam` doesn't take into account any off-diagonal entries in the covariance matrix, but I checked now and it actually does. So I'm a little confused about that too.

We found the issue: In the conversion from `ParticleBeam` to `ParameterBeam`, only the covariances between `x` and `px`, `y` and `py`, and `tau` and `p` are considered. The other diagonal elements of the covariance matrix are currently neglected.

jank324 commented:
> > why does the first call to `plot_2d_distribution` show a beam without any correlations between x and y?
> 
> I was going to say because `ParameterBeam` doesn't take into account any off-diagonal entries in the covariance matrix, but I checked now and it actually does. So I'm a little confused about that too.

@roussel-ryan As @Hespe noted, we figured out that this was in fact a bug and fixed it. I updated the images in the previous comment. Otherwise, does this PR now fix the issue?

jank324 commented:
**Note:** The tests are currently failing because of some `trap: 6` error on the macOS runner that seems to affect all or most PRs and I hope is actually somehow caused by the GitHub Action runners themselves and not by Cheetah.

jank324 commented:
The macOS `abort trap: 6` error is fixed for now.
